firehose.c

```diff
@@ -400,7 +403,16 @@ static int firehose_program(struct qdl_device *qdl, struct program *program, int
 
 	lseek(fd, (off_t) program->file_offset * program->sector_size, SEEK_SET);
 	left = num_sectors;
+	tn = t0;
 	while (left > 0) {
+	        t = time(NULL);
```

Comment:
I was expecting that you just do left % N == 0 for some value of N to avoid printing that often, but I guess this works as well...